tests/test_commands.py

```diff
@@ -234,6 +233,65 @@ def test_nickserv_and_memoserv(alice, bob, wait_until):
     wait_until(lambda: "send Bob hello there\n" in bob.text())
 
 
+@pytest.mark.skipif(
+    os.environ["IRC_SERVER"] == "hircd",
+    reason="hircd doesn't support away notifications",
+)
+def test_away_status(alice, bob, wait_until):
+    alice.entry.insert("end", "/away foo bar baz")
+    alice.on_enter_pressed()
+
+    # Server view
+    wait_until(
+        lambda: "You have been marked as being away\n"
+        in alice.get_server_views()[0].get_text()
+    )
+
+    switch_to_channel_view(alice, "#autojoin")
+
+    # Channel view
+    wait_until(lambda: "You have been marked as being away\n" in alice.text())
+
+    assert "away" in alice.get_current_view().userlist.treeview.item("Alice")["tags"]
+
+    wait_until(
+        lambda: "away" in bob.get_current_view().userlist.treeview.item("Alice")["tags"]
+    )
+
+    alice.entry.insert("end", "/back")
+    alice.on_enter_pressed()
+    wait_until(lambda: "You are no longer marked as being away\n" in alice.text())
+
+    assert (
+        "away" not in alice.get_current_view().userlist.treeview.item("Alice")["tags"]
+    )
+
+
+@pytest.mark.skipif(
+    os.environ["IRC_SERVER"] == "hircd", reason="hircd doesn't support away-notify"
+)
+def test_who_on_join(alice, bob, wait_until):
+    alice.entry.insert(
+        "end", "/part #autojoin"
+    )  # Ensures Alice and Bob are not sharing any channels
```

Comment:
Shouldn't the same exact test pass if they are also both on `#autojoin`?

To run the rest of the test in both ways, you can use e.g. `pytest.mark.parametrize("sharing_channels", [True, False])`. Inside the test, you can then put parting `#autojoin` under `if not sharing_channels:`. Mantaray already has a few tests that use `pytest.mark.parametrize` in this way.

Reply:
Yes, it should pass. I do the `/part` just to make sure the tag can't in some way be taken from another shared channel. Will add the `parametrize`